store/badger.go

```diff
@@ -132,6 +154,39 @@ func (b *badgerKV) CommitWriteBatch(batch WriteBatch) error {
 	return wb.batch.Flush()
 }
 
+func (b *badgerKV) gc(interval time.Duration) {
+	b.closeWg.Add(1)
+
+	go func() {
+		defer b.closeWg.Done()
+
+		ticker := time.NewTicker(interval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-b.closeCh:
+				return
+			case <-ticker.C:
+				for {
+					// Check for close signal, in case it's closed during GC loop.
+					select {
+					case <-b.closeCh:
+						return
+					default:
+					}
+
+					// When there's no more GC, an error will be returned.
+					err := b.db.RunValueLogGC(0.5)
+					if err != nil {
```

Comment:
The `RunValueLogGC` might return an error `ErrNoRewrite` if it didn't perform any rewrite. In this case, you might want to retry the `GC`.

Reply:
Thank you for your help.

I'm sorry, I'm afraid I don't understand.

My understanding is that if the error is nil, we should retry. Otherwise, don't retry.

Reply:
When `RunValueLogGC(..)` is called, it can return
1. `nil` which means GC was successful and a `vlog` file was removed.
2. `ErrNoRewrite` which means it found a `vlog` file to GC but it couldn't GC it because it had less amount of stale data (stale data less than discard ratio)
3. Any other error which occurred while performing garbage collection.

Case 1 is successful GC, case 2 is retry GC and case 3 is something went wrong with GC someone should look at it case.

Reply:
Got it.

In case of `ErrNoRewrite`, we will retry GC in the next cycle.

Currently, when the GC cycle (as for now, it's every 1 minute) is triggered, we would repeatedly call `RunValueLogGC(...)` until an error is returned. 
After which, we would wait until the next GC cycle to repeatedly call `RunValueLogGC(...)` again until an error is returned.

Below is how we handle the cases:
Case 1 (no error): Immediately call `RunValueLogGC(...)` again.
Case 2 (`ErrNoRewrite`): Wait for the next GC cycle to call `RunValueLogGC(...)` again..